### ISSUE TITLE
Accessibility improvement: indicate table heading

### ISSUE THREAD

**Issue by svenaas** [cloud-gov/pages-core#4397]
The published files table at `/sites/[id]/published/[branch]`) is missing a `th` or ARIA attribute property marking the header, which can prevent screen readers from semantically associating cells with the appropriate header.

## Notes

- Reported in February 22, 2024 [Section 508 Evaluation](https://docs.google.com/document/d/18Vl7wCTcCZiXPDribGB_Fo7bnYuIuaJt/edit)

## Acceptance Criteria

- [ ] Add `th` or suitable ARIA property to the table
